catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/util/impl/DescribableServiceMap.java

```diff
@@ -135,4 +154,35 @@ public void clear() {
     public Set<Entry<String, V>> entrySet() {
         return serviceMap.entrySet();
     }
+
+    /**
+     * The handleEvent method takes care of keeping the internal describable map up to date when
+     * service name changes occur.
+     *
+     * @param event
+     */
+    @Override
+    public void handleEvent(Event event) {
+        if (((ServiceEvent) event.getProperty(EventConstants.EVENT)).getType()
+                != ServiceEvent.MODIFIED) {
+            return;
+        }
+        String pid = event.getProperty(Constants.SERVICE_PID)
+                .toString();
+        String oldName = pidToNameMap.get(pid);
+
+        //if we don't have it in our map then it's a service we don't care about
+        if (oldName == null) {
+            return;
+        }
+
+        V service = serviceMap.get(oldName);
+        String newName = service.getId();
```

Comment:
The two maps are always changed together so if we successfully pull the pid from the pidToNameMap then there should be a corresponding entry in the service map. I can add the check if you think it would be safer.